prow/config/config.go

```diff
@@ -1226,7 +1226,8 @@ func validateAgent(v JobBase, podNamespace string) error {
 	b := string(prowapi.KnativeBuildAgent)
 	j := string(prowapi.JenkinsAgent)
 	p := string(prowapi.TektonAgent)
-	agents := sets.NewString(k, b, j, p)
+	x := string(prowapi.JenkinsXAgent)
+	agents := sets.NewString(k, b, j, p, x)
```

Comment:
I would expect more validation here -- should not have a podspec, etc ...

Reply:
@stevekuznetsov Jenkins X does not require the `PodSpec` because every project has a build definition which is stored in the project repository. As an example, you can have a look at this pipeline definition: https://github.com/jenkins-x/jx/blob/master/jenkins-x.yml 

Jenkins X is mainly using the `Refs` and `ExtraRefs` from `ProwJobSpec`.  Given this, I don't think that the config requires extra validation at this point.

Reply:
I think his point is that we should validate that if the agent is JenkinsXAgent that it does not define a podspec?

Reply:
That's good point! I added some checks to validate that no `spec`, `build_spec` or `pipeline_rus_spec` is defined for `jenkins-x` agent.